cmake/templates/BoostConfig.cmake.in

```diff
@@ -51,10 +50,16 @@ if(NOT TARGET "Boost::boost")
       INTERFACE_INCLUDE_DIRECTORIES "${_root_path}/include"
   )
   if(MSVC)
-    set_target_properties(
+    #stop MSVC from trying to auto-link libs
+	set(compile_definitions "BOOST_ALL_NO_LIB=1")
+    if(@BOOST_ALL_DYN_LINK@) #required for MSVC when using shared libs
+	  set(compile_definitions "${compile_definitions} BOOST_ALL_DYN_LINK")
+	endif()
+
+	set_target_properties(
         "Boost::boost"
         PROPERTIES
-        INTERFACE_COMPILE_DEFINITIONS "BOOST_ALL_NO_LIB=1"
+        INTERFACE_COMPILE_DEFINITIONS ${compile_definitions}
```

Comment:
I had missed a file in my test and it was still using set, seems to be fine now.